torrent/torrent_file.py

```diff
@@ -1,64 +1,83 @@
 from torrent.bencoding import Encoder, Decoder
+from torrent.utilities import od_get_key
 import logging
 from enum import Enum
 from hashlib import sha1
 from os.path import isfile
-from collections import OrderedDict
 from urllib.parse import urlsplit
-
+from random import randint
+ 
 logger = logging.getLogger(__name__)
 
 #classes
 
-class TorrentFile(object):
+class TorrentFile():
     def __init__(self, struct_dict):
+        self.multi_file = struct_dict['multi_file']
         self.path = struct_dict['path']
+        self.od_data = struct_dict['orderd_content'] # all the data in OrderdDict object
         self.trackers = struct_dict['trackers']
         self.info_hash = struct_dict['info_hash']
-        self.files = struct_dict['files']
+        if self.multi_file:
+            self.files = struct_dict['files']
         self.length = struct_dict['length']
         self.piece_length = struct_dict['piece_length']
         self.name = struct_dict['name']
+        if type(self.name) == bytes:
+            self.name = self.name.decode('utf-8')
         self.pieces = struct_dict['pieces']
+        
         self.raw_content = struct_dict['raw']
+        self.peer_id = '-PC0001-' + ''.join([str(randint(0, 9)) for _ in range(12)])
+
+    def __repr__(self):
+        return f"torrent_file({self.name})"
 
-class Tracker(object):
+
+class Tracker():
     #TODO think if this is the right place hold this object and if the the future the tracker going to
     #manage the connection
     """
     the object contain 2 types of trackers udp or http trackers
     url(domain):str, path:str, type:'http' or 'udp, port:'int'
     """
+    def __init__(self, tracker_type, url, path,  port=80):
+        self.schema = tracker_type
+        self.url = url
+        self.path = path
+        self.port = port
+
+        if tracker_type == 'https':
```

Comment:
`self.tracker_type = tracker_type.replace('https', 'http')` (if that doesn't work use re.sub, or use an enum)